tests/conftest.py

```diff
@@ -789,3 +791,16 @@ def with_model_id(event: Event, model_id: Text) -> Event:
 @pytest.fixture(autouse=True)
 def sanic_test_mode(monkeypatch: MonkeyPatch):
     monkeypatch.setattr(Sanic, "test_mode", True)
+
+
+def filter_expected_warnings(records: WarningsRecorder) -> WarningsRecorder:
+    records_copy = copy.deepcopy(records.list)
```

Comment:
not sure why we're deep-copying the list, as we're just iterating on it, and we anyway pop from the `records` object at line 804

Reply:
If I didn't deepcopy the list, and iterated over the original list, I'd be popping elements from the list I'm iterating over, and the looping would finish before all the warnings that need to be removed are removed, so `records` would still contain unfiltered `DeprecationWarnings`.

Reply:
🤦🏻 yes, indeed 💤